lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,307 @@
+			var err error
+			minerAddr, err = net.Miner.NewAddress()
+			if err != nil {
+				ht.Fatalf(
+					"unable to create new miner address: %v",
+					err,
+				)
+			}
+		}
+
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		if _, err := carol.SendCoins(ctxt, &lnrpc.SendCoinsRequest{
+			Addr:    minerAddr.String(),
+			SendAll: true,
+		}); err != nil {
+			ht.Fatalf("unable to sweep coins from node %v: %v",
+				node.Name(), err)
+		}
+
+		// Ensures we don't leave any transaction in the mempool after sweeping.
+		mineBlocks(ht, net, 1, 1)
+
+		// Make sure all funds were swept.
+		checkChannelBalance(node, 0)
+	}
+
+	runTestCase := func(carol, dave *lntest.HarnessNode,
+		testCase *chanFundMaxTestCase) func(t *testing.T) {
+
+		return func(t *testing.T) {
+			sendCoinsAndAssert(testCase.amount, carol)
+			defer func() {
+				if testCase.amount <= 2000 {
+					// Add additional funds to sweep "dust" UTXO.
+					sendCoinsAndAssert(
+						btcutil.Amount(100000), carol,
+					)
+				}
+
+				// Remove all funds from Carol.
+				sweepNodeWalletAndAssert(carol)
+			}()
+
+			// The parameters to try opening the channel with.
+			chanParams := lntest.OpenChannelParams{
+				Amt:        0,
+				PushAmt:    testCase.pushAmt,
+				SatPerByte: testCase.feeRate,
+				FundMax:    true,
+			}
+
+			// If we don't expect the channel opening to be
+			// successful, simply check for an error.
+			if testCase.chanOpenShouldFail {
+
+				ctxt, _ = context.WithTimeout(
+					ctxb, channelOpenTimeout,
+				)
+
+				_, err = net.OpenChannel(
+					ctxt, carol, dave, chanParams,
+				)
+
+				if err == nil {
+					t.Errorf("should not open channel "+
+						"for: %v", testCase.name)
+				}
+
+				return
+			}
+
+			// Otherwise, if we expect to open a channel use the helper function.
+			ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+			chanPoint := openChannelAndAssert(
+				ctxt, ht, net, carol, dave, chanParams,
+			)
+			defer func() {
+				// Close the channel between Carol and Dave, asserting
+				// that the channel has been properly closed on-chain.
+				ctxt, _ = context.WithTimeout(
+					ctxb, channelCloseTimeout,
+				)
+
+				closeChannelAndAssert(
+					ctxt, ht, net, carol, chanPoint, false,
+				)
+			}()
+
+			cType, err := channelCommitType(carol, chanPoint)
+			if err != nil {
+				ht.Fatalf("unable to get channel type: %v", err)
+			}
+
+			// Carol's balance should be her amount subtracted by the
+			// commitment transaciton fee.
+			err = checkChannelBalance(
+				carol, testCase.carolBalance-cType.calcStaticFee(0),
+			)
+			if err != nil {
+				t.Errorf("carol's balance is wrong: %v", err)
+			}
+
+			// Ensure Dave's balance within the channel is equal to the push amount.
+			err = checkChannelBalance(dave, testCase.pushAmt)
+			if err != nil {
+				t.Errorf("dave's balance is wrong: %v", err)
+			}
+		}
+	}
+
+	var testCases = []*chanFundMaxTestCase{
+		{
+			name:               "wallet amount is dust",
```

Comment:
tbh this looks more like checks that should be done on the unit test level...

Reply:
You are thinking something like a test `TestFundingManagerFailDust` inside `fundingmanager_test.go`?

Reply:
I'm saying that all these tests don't really belong on the integration test level, since they test very specific behaviour of a particular unit (the coin selection).

Reply:
The test cases that are more specific for the `fundmax` flag are indeed already tested for inside `coin_select_test.go` (dust values, min/max channel sizes...).

However, as far as I can tell there does not exist integration tests for the `openchannel` command. Do you think it would make sense to later add a PR with only the integration tests for the `openchannel` command?